src/amqp_definitions.c

```diff
@@ -382,8 +382,8 @@ int error_get_condition(ERROR_HANDLE error, const char** condition_value)
                 }
                 else
                 {
-                    int get_single_value_result;
-                    if ((get_single_value_result = amqpvalue_get_symbol(item_value, condition_value)) != 0)
+                    int get_single_value_result = amqpvalue_get_symbol(item_value, condition_value);
```

Comment:
Amqp_definitions is a generated file. It is generated by running uamqp_generator which lives in the uamqp repo. It is a .Net project using T4 to generate the code. Basically running the .Net project will just generate the .h and .c files where they should be.

Reply:
Where is this uamqp repo? According to [GitHub](https://github.com/search?q=uamqp&type=Repositories) it doesn't exist. If I understand T4 Templates correctly, I hoping I can open an issue there to fix the template to generate better code. The pattern of 

```
int foo;
if ((foo = bar()) != baz) 
```

is needlessly complicated depending on a gui based debugger to even derive the debuggability value from the pattern. Better would be either

```
if (baz != bar())
```
...or...
```
int foo = bar();
if (foo != baz)
```

Reply:
Sorry, I'll take a look at the code in the morning. I'd love to take a swing at T4 templates.

Reply:
That was super fun. Looks like the generator hasn't been updated since amqp was split out of the sdk. I can understand now how the generated code got that way. A meta code optimization produces undesirable results. Worth pondering.

Reply:
I was going to suggest adding the following at the top of the .tt files

```
// GENERATED FILE - DO NOT CHANGE THIS FILE
// This file is generated by the project in the uamqp_generator folder.
// Please make your changes there.
// GENERATED FILE - DO NOT CHANGE THIS FILE
```

but that generator generates a lot of files and that would be a big delta from this PR

Reply:
I actually just added this yesterday because it was not obvious :-). This is the sha. [https://github.com/Azure/azure-uamqp-c/commit/3c74ce096e9a92473076615ca11becb60ffbc511](https://github.com/Azure/azure-uamqp-c/commit/3c74ce096e9a92473076615ca11becb60ffbc511)

It might be that you are on a slightly older version, but that should be fine.

I will merge your PR today. Thanks for the contribution!

---
In reply to: [188734601](https://github.com/Azure/azure-uamqp-c/pull/227#discussion_r188734601) [](ancestors = 188734601)